ipc/source/GGCoreIpcClient.cpp

```diff
@@ -0,0 +1,508 @@
+            {
+            }
+
+            void SubscriptionResponseMessage::s_customDeleter(SubscriptionResponseMessage *response) noexcept
+            {
+                OperationResponse::s_customDeleter((OperationResponse *)response);
+            }
+
+            GreengrassIpcClient::GreengrassIpcClient(
+                ConnectionLifecycleHandler &lifecycleHandler,
+                Crt::Io::ClientBootstrap &clientBootstrap,
+                Crt::Allocator *allocator) noexcept
+                : m_connection(lifecycleHandler, allocator), m_clientBootstrap(clientBootstrap), m_allocator(allocator)
+            {
+                aws_event_stream_library_init(m_allocator);
+
+                m_greengrassModelRetriever.m_ModelNameToSoleResponseMap[Crt::String("aws.greengrass#PublishToTopic")] =
+                    PublishToTopicResponse::s_loadFromPayload;
+                m_greengrassModelRetriever
+                    .m_ModelNameToSoleResponseMap[Crt::String("aws.greengrass#SubscribeToTopic")] =
+                    SubscribeToTopicResponse::s_loadFromPayload;
+                m_greengrassModelRetriever
+                    .m_ModelNameToStreamingResponseMap[Crt::String("aws.greengrass#SubscribeToTopic")] =
+                    SubscriptionResponseMessage::s_loadFromPayload;
+            }
+
+            std::future<EventStreamRpcStatus> GreengrassIpcClient::Connect(
+                ConnectionLifecycleHandler &lifecycleHandler,
+                const Crt::Optional<Crt::String> &ipcSocket,
+                const Crt::Optional<Crt::String> &authToken) noexcept
+            {
+                std::promise<EventStreamRpcStatus> initializationPromise;
+
+                Crt::String finalIpcSocket;
+                if (ipcSocket.has_value())
+                {
+                    finalIpcSocket = ipcSocket.value();
+                }
+                else
+                {
+                    finalIpcSocket = Crt::String(std::getenv("AWS_GG_NUCLEUS_DOMAIN_SOCKET_FILEPATH_FOR_COMPONENT"));
+                }
+
+                Crt::String finalAuthToken;
+                if (authToken.has_value())
+                {
+                    finalAuthToken = authToken.value();
+                }
+                else
+                {
+                    finalAuthToken = Crt::String(std::getenv("SVCUID"));
+                }
+
+                /* Encode authToken as JSON. */
+                Crt::StringStream authTokenPayloadSS;
+                authTokenPayloadSS << "{\"authToken\":\"" << finalAuthToken << "\"}";
+
+                if (!m_clientBootstrap)
+                {
+                    initializationPromise.set_value({EVENT_STREAM_RPC_INITIALIZATION_ERROR, 0});
+                    return initializationPromise.get_future();
+                }
+
+                Crt::Io::SocketOptions socketOptions;
+                socketOptions.SetSocketDomain(Crt::Io::SocketDomain::Local);
```

Comment:
It might be a good idea to have a Config class so that we won't need to introduce API changes if connect requires a different set of parameters in the future